### ISSUE TITLE
Favicons don't show up in Firefox Vomnibar

### ISSUE THREAD

**Issue by nsunderland1** [philc/vimium#3674]
**Describe the bug**
When viewing the current tabs in the Vomnibar in Firefox, favicons don't show up next to the entries.

Behaviour in Firefox:

![image](https://user-images.githubusercontent.com/16548079/94186884-39a58900-fe75-11ea-9ab5-1aec1a2bd07b.png)

Expected behaviour (Chrome):

![image](https://user-images.githubusercontent.com/16548079/94186766-17137000-fe75-11ea-9d30-0ef16123ae3c.png)

**To Reproduce**
1. Open any page in Firefox
2. Press Shift-T to open the list of current tabs in the Vomnibar.
3. The favicon for that tab doesn't appear next to its Vomnibar entry.

This also occurs with other Vomnibar shortcuts (shift-O, shift-B...). Tried on a clean Firefox install with no tweaks to settings or additional addons, no difference.

**Browser and Vimium version**

Browser: Firefox 80.0.1
OS: Linux 5.4.0-48-generic #52-Ubuntu SMP (Ubuntu 20.04.1 LTS)
Vimium version: 1.66

**Comment by nsunderland1** [comment]
I see this is by design:

https://github.com/philc/vimium/blob/bdf654aebe6f570f427c5f7bc9592cad86e642b5/background_scripts/completion.js#L62-L63

In #2878 the decision was made to use the Google favicon resolver to implement this feature. I can't find much information about the Google favicon resolver, but the [article](https://web.archive.org/web/20090303215925/http://codingclues.eu/2009/retrieve-the-favicon-for-any-url-thanks-to-google/) cited in that PR provides a way of using it in browsers other than Chrome using `http://www.google.com/s2/favicons?domain_url=[url]`. This seems to give mixed results in Firefox:

![image](https://user-images.githubusercontent.com/16548079/94194941-657a3c00-fe80-11ea-8af3-ff7f94c7e2dd.png)

Obviously it won't work for `about:` pages, not sure what the issue with Github here is though.

**Comment by kiuKisas** [comment]
Can we just load the favicon with ${base_url}/favicon.ico ? As it suppose to be there..

**Comment by tommyalatalo** [comment]
Any progress on this? Just found this today.

**Comment by gdh1995** [comment]
The `favIconUrl` is a nightmare: they're so long that it affects loading time of Vomnibar if Vimium tells Firefox to read, send, insert, parse and show it - https://github.com/gdh1995/vimium-c/issues/365#issuecomment-873785756

**Comment by jgoedde** [comment]
Anything new here? I just stumbled on this today.

**Comment by mijoharas** [comment]
fyi, was just looking around for this. As far as I can tell the performance bug in firefox has been closed? I think we could possibly add this for firefox now.

In addition. I was reading up on the `favicon` permission in chrome, and I don't see it saying it's only for tabs. It appears to be any favicon that's cached. Have we tried displaying tabs for history items?

**Comment by Igetin** [comment]
I already replied to an older similar issue (#2424), but I'll say it here as well:

I'm currently on Chrome but trying to move to Firefox. Chrome has a neat integrated tab search UI with a keyboard shortcut, and it shows the favicons. That visual cue makes scanning and picking the right tab much quicker.

Firefox does not have an equivalent built-in tab search. Vimium's tab search is the closest alternative workflow I have found, but without favicons the UX is noticeably worse.

**Comment by ckardaris** [comment]
I found this thread searching for favicon support in `vimium` myself and I wanted share something for those not aware.

> Firefox does not have an equivalent built-in tab search. Vimium's tab search is the closest alternative workflow I have found, but without favicons the UX is noticeably worse.

`firefox` has support for tab search: https://support.mozilla.org/en-US/kb/search-open-tabs-firefox.

It does require some additional key presses, but `Ctrl-L` + `%` + `Space` is not that bad.
Of course `vimium`'s `T` is faster and it would be good to have icon support if the implementation is reasonable.

**Comment by Igetin** [comment]
> > Firefox does not have an equivalent built-in tab search. Vimium's tab search is the closest alternative workflow I have found, but without favicons the UX is noticeably worse.
> 
> `firefox` has support for tab search: https://support.mozilla.org/en-US/kb/search-open-tabs-firefox.
> 
> It does require some additional key presses, but `Ctrl-L` + `%` + `Space` is not that bad. Of course `vimium`'s `T` is faster and it would be good to have icon support if the implementation is reasonable.

Those extra presses add up when you use the shortcut dozens of times daily. Chrome's integrated tab search is much more intuitive: a single shortcut (Ctrl+Shift+A) brings up the tab search. In addition, the best match is **automatically** highlighted based on the input. With Firefox's integrated tab search, you always have to **manually** highlight the item you want, and only then can you press Enter to activate it. Vimium also has the same auto-highlight feature, which is why I'm relying on it instead of using Firefox's integrated tab search.

**Comment by ckardaris** [comment]
> Those extra presses add up when you use the shortcut dozens of times daily. Chrome's integrated tab search is much more intuitive: a single shortcut (Ctrl+Shift+A) brings up the tab search. In addition, the best match is **automatically** highlighted based on the input. With Firefox's integrated tab search, you always have to **manually** highlight the item you want, and only then can you press Enter to activate it. Vimium also has the same auto-highlight feature, which is why I'm relying on it instead of using Firefox's integrated tab search.

I agree. This is why I am not using the feature myself. Just wanted to set the record straight for people who may not know about this feature. It is technically equivalent to what Chrome offers. It's just not good enough in comparison.

**Comment by Igetin** [comment]
Yes, I should've specified that the _UX_ of the feature is not equivalent, not that the feature itself is not equivalent. 🙂
